core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -544,58 +544,82 @@ private[master] class Master(
    * has enough cores and memory. Otherwise, each executor grabs all the cores available on the
    * worker by default, in which case only one executor may be launched on each worker.
    */
-  private def startExecutorsOnWorkers(): Unit = {
-    // Right now this is a very simple FIFO scheduler. We keep trying to fit in the first app
-    // in the queue, then the second app, etc.
+
+  private[master] def scheduleExecutorsOnWorkers(app: ApplicationInfo,
+    usableWorkers: Array[WorkerInfo], spreadOutApps: Boolean): Array[Int] = {
+    val coresPerExecutor = app.desc.coresPerExecutor.getOrElse(1)
```

Comment:
then we should add a comment here to say that:
```
// If the number of cores per executor is not specified, then we can just schedule
// 1 core at a time since we expect a single executor to be launched on each worker
```